Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py

```diff
@@ -66,14 +66,26 @@ def from_string(self, string):
         return None
 
     def credentials(self, required=True):
-        return credentials(
+        args = dict(
             url=self.api_url,
             required=required,
             name=self.url.split('/')[2].replace('.', '_').upper(),
-            prompt="GitHub's API\nPlease generate a 'Personal access token' via 'Developer settings' for your user",
+            prompt='''GitHub's API
+Please generate a 'Personal access token' via 'Developer settings' with 'repo' and 'workflow'
+access for your {host} user
+https://{host}/settings/tokens'''.format(host=self.url.split('/')[2]),
             key_name='token',
         )
 
+        username, token = credentials(**args)
+        if username and '@' in username:
+            sys.stderr.write("Provided username contains an '@' symbol, which is not allowed in GitHub\n")
+            delete_credentials(url=args['url'], name=args['name'])
+            username, token = credentials(**args)
+        if username and '@' in username:
+            raise ValueError("GitHub usernames cannot have '@' in them")
```

Comment:
Nit: this is bailing out, right? If so, can print a message asking user to re-run this command again later-on.

Reply:
This _probably_ results in us bailing out, but we don't actually know the command being run. Ideally, it's `git-webkit setup`, but `git-webkit pr` also uses this code-path, as does `git-webkit find` if the repository in question declared that credentials are required (as it might if, for example, the repository was not a public one). It's probably better to have the caller handle this error than to print something out here because the caller will know if the command is safe to re-run

Reply:
ok, sounds good